storage/remote/client.go

```diff
@@ -200,6 +200,9 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 	if httpResp.StatusCode/100 == 5 {
 		return RecoverableError{err}
 	}
+	if httpResp.StatusCode/100 == 4 && httpResp.StatusCode%100 == 29 {
```

Comment:
You don't need that complexity

```suggestion
	if httpResp.StatusCode == 429 {
```